www/drivers/VCNL4010.py

```diff
@@ -0,0 +1,73 @@
+"""
+Helper modole for the VCNL4010 IR distance and ambient light sensor.
+
+Adapted from:
+https://www.controleverything.com/content/Light?sku=VCNL4010_I2CS#tabs-0-product_tabset-2
+"""
+
+try:
+    import smbus
+except ImportError:
+    pass
+
+from MockSmbus import MockSmbus
+
+
+class VCNL4010:
+    """A module to set up and read from a VCNL4010."""
+
+    def __init__(self, i2c_port, i2c_addr,
+                 led_current=None, binary_threshold=None):
+        """Create a VCNL4010 driver module."""
+        if not (led_current and binary_threshold):
+            led_current = 0x0A
+            binary_threshold = 2150
+        self.binary_threshold = binary_threshold
+
+        print "Setting up VCNL4010 with LED current {} and threshold {}"\
+            .format(led_current, self.binary_threshold)
+
+        try:
+            self.bus = smbus.SMBus(i2c_port)
+        except (IOError, NameError):
+            print "VCNL4010 was unable to connect to I2C bus {}. " \
+                  "Mocking out the bus".format(i2c_port)
+            self.bus = MockSmbus()
+
+        self.i2c_addr = i2c_addr
+
+        # command register, 0x80
+        # 0xFF(255) Enable ALS and proximity measurement, LP oscillator
+        self.bus.write_byte_data(i2c_addr, 0x80, 0xFF)
```

Comment:
What does the real SMBus do if there is a nak? Do you need to handle an exception or error return value?

Reply:
Good call. They throw IOError exceptions. I added a handler that just prints an error message and continues. Eventually, we'll come up with a red light or something for the rover to show that something's not working.